TopReadWidget/WMFTodayTopReadWidgetViewController.swift

```diff
@@ -347,33 +347,41 @@ class WMFTodayTopReadWidgetViewController: UIViewController, NCWidgetProviding {
         fetch(siteURL: siteURL, date:Date(), attempt: 1, completionHandler: completionHandler)
     }
     
-    func updateUIWithTopReadFromContentStoreForSiteURL(siteURL: URL, date: Date) -> Bool {
+    func updateUIWithTopReadFromContentStoreForSiteURL(siteURL: URL, date: Date) -> NCUpdateResult {
         if let topRead = self.userStore.viewContext.group(of: .topRead, for: date, siteURL: siteURL) {
             if let content = topRead.content as? [WMFFeedTopReadArticlePreview] {
+                if let previousGroupKey = self.group?.key,
+                    let topReadKey = topRead.key,
+                    self.results.count > 0,
+                    topReadKey == previousGroupKey {
+                    return .noData
```

Comment:
Thoughts on using `.dataUnchanged` instead of `.noData`? (just based on the commit message)